tests/test_filters.py

```diff
@@ -739,6 +739,16 @@ def test_update_type_edited_channel_post(self, update):
         assert Filters.update.channel_posts(update)
         assert Filters.update(update)
 
+    def test_update_type_poll(self, update):
+        update.poll = True
+        assert Filters.update.poll(update)
+        assert Filters.update(update)
```

Comment:
Why does this work? By this [line](https://github.com/python-telegram-bot/python-telegram-bot/blob/d96d233dc5d74a21b800ac970d6d2ef431cc58cb/telegram/ext/filters.py#L1056) I'd expect that only (edited) messages/channel posts come through …

Reply:
I override this for poll update type. I think.

Reply:
You override `filter` for the poll type, yes, but `Filters.update(update)` calls `Filters.update.filter()`. But I found the problem: `update` has a `message` attribute (see line 30 above). My suggestion would be:

1. edit `Filters.update.filter` to also allow poll type (i.e. add `or self.poll(update)`)
2. in `test_update_type_poll` and `test_update_type_poll_answer` set `update.message = None` to make sure the poll atrribute is the one getting tested

Reply:
do we actually want that though? Or would we prefer a PollHandler? Only message and channel posts are handled in the MessageHandler so far.

Reply:
_If_ we do, here are some thoughts:

* We'd need both PollHandler and PollAnswerHandler
* or/and maybe a keyword poll_answer_updates to choose if PollHandler should also handler PollAnswers
* PollHandler should handle updates with update.message.poll imo
* A pattern argument could filter for poll descriptions